jsonschema/scripts/templates/dotnet.dotnet.erb

```diff
@@ -0,0 +1,150 @@
+<%- @schemas.each do |key, schema| -%>
+<%= class_name(key) -%>.cs
+using System;
+using System.Collections.Generic;
+
+// ------------------------------------------------------------------------------
+// This code was generated based on the Cucumber JSON schema
+// Changes to this file may cause incorrect behavior and will be lost if 
+// the code is regenerated.
+// ------------------------------------------------------------------------------
+
+namespace Io.Cucumber.Messages.Types;
+
+/**
+ * Represents the <%= class_name(key) %> message in Cucumber's message protocol
+ * @see <a href="https://github.com/cucumber/messages" >Github - Cucumber - Messages</a>
+ <%- if not (schema['description'] || []).empty? -%>
+ *
+<%= format_description(schema['description'], indent_string: '') %>
+<%- end -%>
+ */
+
+public sealed class <%= class_name(key) %> 
+{
+  <%- schema['properties'].each do |property_name, property| 
+      required = (schema['required'] || []).index(property_name) 
+      isValueType = type_for(class_name(key), property_name, property) == "long" || type_for(class_name(key), property_name, property) == "bool"
```

Comment:
this logic feels quite complex and can mess with the ordering of the lines. Is there any way we can push this into the class proper?

Reply:
This cannot be pushed into the .NET class. The intent is to identify those properties and fields that are optional value types. The declarations for these need to be different than a regular long or bool; they are declared as Nullable<T> (see lines 35/37 and 65/67 for a comparison). 
I don't understand your meaning of "can mess with the ordering of the lines" - what is the impact? Is there an example within the code to demonstrate?

Reply:
Sorry I meant more migrating this into the `Dotnet` ruby class that would determine this logic so we can make the overall erb template smaller. But I think this is probably best done after this has been merged and then I tag you for review. It will probably make more sense that way and you've done a lot of work here I don't want to delay / overcomplicate things.